### PR TITLE
feat: update spfx samples for mgt v4

### PR DESCRIPTION
**Pull request by gavinbarron**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/mgt-samples/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | yes                               |
| New sample?     | no                                |
| Related issues? | fixes #X, partially #Y, mentioned in #Z |

## What's in this Pull Request?

Removes dynamic import in sp-mgt 
Updates gulp files to patch webpack config to handle new lit dependencies

### CODE REVIEW

File: samples/app/sp-mgt/package.json [hunk 1]
```diff
@@ -12,9 +12,9 @@
   },
   "dependencies": {
     "tslib": "2.3.1",
-    "@microsoft/mgt-element": "*",
-    "@microsoft/mgt-components": "*",
-    "@microsoft/mgt-sharepoint-provider": "*",
+    "@microsoft/mgt-element": "3.1.3-preview.ec25e57",
+    "@microsoft/mgt-components": "3.1.3-preview.ec25e57",
+    "@microsoft/mgt-sharepoint-provider": "3.1.3-preview.ec25e57",
```
Comment (gavinbarron) [inline samples/app/sp-mgt/package.json]:
```suggestion
    "@microsoft/mgt-element": "*",
    "@microsoft/mgt-components": "*",
    "@microsoft/mgt-sharepoint-provider": "*",
```

File: samples/app/sp-webpart/package.json [hunk 2]
```diff
@@ -14,10 +14,10 @@
...
   },
   "dependencies": {
-    "@microsoft/mgt-element": "*",
-    "@microsoft/mgt-react": "*",
-    "@microsoft/mgt-spfx-utils": "*",
-    "@microsoft/mgt-sharepoint-provider": "*",
+    "@microsoft/mgt-element": "3.1.3-preview.ec25e57",
+    "@microsoft/mgt-react": "3.1.3-preview.ec25e57",
+    "@microsoft/mgt-spfx-utils": "3.1.3-preview.ec25e57",
+    "@microsoft/mgt-sharepoint-provider": "3.1.3-preview.ec25e57",
```
Comment (gavinbarron) [inline samples/app/sp-webpart/package.json]:
```suggestion
    "@microsoft/mgt-element": "*",
    "@microsoft/mgt-react": "*",
    "@microsoft/mgt-spfx-utils": "*",
    "@microsoft/mgt-sharepoint-provider": "*",
```

**Review by gavinbarron — COMMENTED** [review]
Noting changes that will need to be made after MGT v4 releases and before merging.

File: samples/app/sp-mgt/gulpfile.js [hunk 3]
```diff
@@ -1,16 +1,45 @@
...
   var result = getTasks.call(build.rig);
 
-  result.set('serve', result.get('serve-deprecated'));
+  result.set("serve", result.get("serve-deprecated"));
 
   return result;
 };
 
-build.initialize(require('gulp'));
+// add babel-loader and some transforms to handle es2021 language features which are unsupported in webpack 4 by default
```
Comment (sebastienlevert) [inline samples/app/sp-mgt/gulpfile.js:19]: Wow this is wild. Can we package this setup in `mgt-spfx-utils` to make it reusable without handcrafting this code? And if not, at least make it obvious in our SPFx specific documentation

**Review by sebastienlevert — CHANGES_REQUESTED** [review]

File: samples/app/sp-mgt/gulpfile.js (hunk 3 above)
Comment (gavinbarron) [reply samples/app/sp-mgt/gulpfile.js:19]: We don't gain enough given the changes we'd be making to provide the function, which then needs to be dynamically imported thanks to esm/cjs differences

**Comment by sebastienlevert** [comment]
@gavinbarron let's update this one with the latest discussions with path on Windows. Then I think we're good to go for merging.

**Review by sebastienlevert — CHANGES_REQUESTED** [review]
@gavinbarron let's update this one with the latest discussions with path on Windows. Then I think we're good to go for merging.

**Comment by gavinbarron** [comment]
@sebastienlevert I've updated the gulpfiles to use `path.sep` so I think this is good to go

**Review by sebastienlevert — APPROVED** [review]

**Pull request merged**